torchao/prototype/mx_formats/mx_ops.py

```diff
@@ -80,12 +80,26 @@ def _get_gemm_choice(
 
 
 def _addmm_mx_dispatch(
-    a: MXTensor, b: MXTensor, aten_op, bias: Optional[torch.Tensor] = None
+    a: torch.Tensor, b: MXTensor, aten_op, bias: Optional[torch.Tensor] = None
 ) -> torch.Tensor:
     """
     Core implementation shared between mx_mm and mx_addmm.
     The only difference is whether bias is None or not.
     """
+
+    if not isinstance(a, MXTensor):
+        assert b.act_quant_kwargs is not None, "weight-only quant not yet supported"
+        k = b.act_quant_kwargs
+        a = MXTensor.to_mx(
```

Comment:
nit: rename to `from_hp` for all tensor subclasses for consistency (feel free to do in separate PR)